src/main/webapp/gameHelper.js

```diff
@@ -23,20 +23,31 @@ var experience;
 var threshold;
 // For now, this is hard coded, will change with new servlet
 // https://github.com/googleinterns/step144-2020/issues/171
-var newThreshold = 20;
- 
+var thresholdIncrement;
+
 function getDialogue() {
   const responsePromise = fetch('/game-dialogue');
   responsePromise.then(handleResponse);
 }
  
 function handleResponse(response) {
-  const textPromise = response.json();
+  const textPromise = response.text();
   textPromise.then(addDialogueToDom);
 }
  
-function addDialogueToDom(dialogue) {
+function addDialogueToDom(textResponse) {
+  responseArray = textResponse.split("\n");
+  // Divides the response into level and dialogue
+  var levelString = responseArray[0];
```

Comment:
how are you getting levelString if dialogue never included an integer value? will the  static data we have change?

Reply:
or will you be returning the current gamestage id and the dialogue in the java servlet

Reply:
That's the change I made in GameDialoge.java. Instead of the json single object of dialogue, the response now includes the gameStage id as a string  and the dialogue as a text response, so it's fetched as a long string, split by a new line.